packages/next-on-pages/src/buildApplication/processVercelFunctions/configs.ts

```diff
@@ -61,6 +68,13 @@ export async function collectFunctionConfigsRecursively(
 	return configs;
 }
 
+function checkPrerenderConfigExists(funcPath: string, files: PathInfo[]) {
+	return files.find(
+		({ path }) =>
+			path === funcPath.replace(/\.func$/, '.prerender-config.json'),
+	);
```

Comment:
the replace here is applied for every item in the iteration, I think it'd be better to have it outside of the `find`

```suggestion
    const targetPath = funcPath.replace(/\.func$/, '.prerender-config.json'); 
	return files.find(
		({ path }) =>
			path === targetPath,
	);
```

Reply:
PS: I don't remember too well the paths here, could you remind me why `.func$` is there and needs to be removed? 😅

Reply:
Sure, so it would be a list of paths like this:

<img width="262" alt="image" src="https://github.com/cloudflare/next-on-pages/assets/10815538/c6c36bf9-9648-4183-af89-5a0c250f5d22">

We need to find if there is a prerender config for the function directory if it is an `API` route, since this is the only way it appears to tell if it was prerendered or not. So we just replace the .func with the name of the configs to check it exists or not.

Reply:
ahhhh, perfect, got it! thanks a lot! ❤️